test/tools/router_check/router.cc

```diff
@@ -394,6 +394,10 @@ bool RouterCheckTool::compareRedirectPath(ToolConfig& tool_config, const std::st
   if (tool_config.route_->directResponseEntry() != nullptr) {
     if (!headers_finalized_) {
       tool_config.route_->directResponseEntry()->rewritePathHeader(*tool_config.headers_, true);
+
+      // The real router makes use of sendLocalReply, which automatically sets the content-type header to text/plain.
+      // Imitate that behavior here.
+      tool_config.headers_->setReferenceContentType(Http::Headers::get().ContentTypeValues.Text);
```

Comment:
Nice catch! My understanding is that https://github.com/envoyproxy/envoy/blob/10d40f51e3e223153d9e9286b2b784b3c2e111b8/source/common/http/utility.cc#L337 is called only when there is no message body, and I confirmed by curling some direct response routes that only return a status code and checking that there was no content-length header set. Could you incorporate that logic (as well as perhaps setting the content-length header while you're making changes) and add/edit unit tests to demonstrate that?

Reply:
One thing to think about here is whether we can share production code from the router in this tool, to avoid these types of issues in the future?

Reply:
I'm using the production `sendLocalReply` now.  It's a step in the right direction.